vnext/Desktop.DLL/react-native-win32.x64.def

```diff
@@ -63,6 +63,7 @@ EXPORTS
 ?GetRuntimeOptionInt@React@Microsoft@@YA?BHAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z
 ?makeChakraRuntime@JSI@Microsoft@@YA?AV?$unique_ptr@VRuntime@jsi@facebook@@U?$default_delete@VRuntime@jsi@facebook@@@std@@@std@@$$QEAUChakraRuntimeArgs@12@@Z
 ?Make@IHttpResource@React@Microsoft@@SA?AV?$unique_ptr@UIHttpResource@React@Microsoft@@U?$default_delete@UIHttpResource@React@Microsoft@@@std@@@std@@XZ
+?MakeNapiJsiRuntime2@JSI@Microsoft@@YA?AV?$unique_ptr@VRuntime@jsi@facebook@@U?$default_delete@VRuntime@jsi@facebook@@@std@@@std@@PEAUnapi_env__@@@Z
```

Comment:
>MakeNapiJsiRuntime2 [](http://example.com/codeflow?start=1&length=19)

Ideally we must not export this function, but instead add the source files to the Nuget package for other project to compile them outside of the ABI boundary.

Reply:
We will remove it eventually, but we don't have a test project that:
1. Uses GoogleTest.
2. Statically links the library that includes `NapiJsiRuntime.cpp`.

The test is implemented in an integration test project (meaning, it consumes the DLL and only has access to its exported symbols).

Reply:
Yes, these are new files and we need to modify an existing unit test project to do that. The main point is that the NapiJsiRuntime.h and NapiJsiRuntime.cpp must be compiled as a part of the project that uses the RN DLL.
The ideal place for these files to satisfy these needs in the Microsoft.ReactNative.Cxx project where we have other .h/.cpp files which we want to be compiled outside of the DLL to use the ABI-safe APIs.

Reply:
I'll look into both aspects after merging this case, but I don't know whether `Microsoft.ReactNative.Cxx` can be consumed by the Desktop DLL.